app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,51 @@
+package de.xikolo.storages
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.storages.base.BaseStorage
+
+typealias RecentCourse = Pair<String, String>
+
+const val MAX_SHORTCUTS = 4
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    companion object {
+        private const val PREF_RECENT_COURSES = "preference_recent_courses"
+
+        fun addCourseToRecentCourses(
+            courseId: String,
+            title: String,
+            courses: LinkedHashSet<RecentCourse>
+        ): LinkedHashSet<RecentCourse> {
```

Comment:
Does ist specifically have to be a `LinkedHashSet` or would be `Set` sufficient?

Reply:
If you need an ordered collection, use `List` instead

Reply:
We need order and uniqueness of entries, therefore the usage of `LinkedHashSet`.